PolyEngine/Core/Src/pe/core/utils/Result.hpp

```diff
@@ -10,20 +10,39 @@ namespace pe::core::utils
 	{
 	public:
 		Result() = delete;
-		Result(Result&) = delete;
+		Result(Result&) = default;
 		Result(Result&&) = default;
-		Result& operator=(Result&) = delete;
+		Result& operator=(Result&) = default;
 		Result& operator=(Result&&) = default;
 		Result(T okVal) : m_value(std::move(okVal)) {}
 		Result(E errVal) : m_value(std::move(errVal)) {}
 		
-		bool IsOk() const { return m_value.index() == 0; }
-		bool IsErr() const { return !IsOk(); }
-		T& GetValue() { return std::get<T>(m_value); }
-		E& GetError() { return std::get<E>(m_value); }
-		const T& GetValue() const { return std::get<T>(m_value); }
-		const E& GetError() const { return std::get<E>(m_value); }
-	
+		bool isOk() const { return m_value.index() == 0; }
+		bool isErr() const { return !isOk(); }
+		
+		T& getValue() { return std::get<T>(m_value); }
+		E& getError() { return std::get<E>(m_value); }
+		const T& getValue() const { return std::get<T>(m_value); }
+		const E& getError() const { return std::get<E>(m_value); }
+
+		template <typename U>
+		Result<U, E> and(Result<U, E> res)
```

Comment:
Why not make those functions standalone and variadic?

Reply:
I suggest simply using `operator&` and `operator|`.

Reply:
I suggest instead to define for all operators (let's use `+` for example here)
`template <typename X1, typename X2, typename Y, typename E> R<Y,E> operator+(R<X1,E> x1, R<X2,E> x2)`
returning error of `x1` or error of `x2` or `x1 + x2`

then, as mentioned before
`template <typename T, typename E> R<T, E> first(R<T, E> ... xs)`
and overload for ranges
`template <typename T, typename E, template<typename> typename C> R<T,E> first(C<R<T,E>>)`

Reply:
or perhaps ditch `first`/`last`, since those functions discard potentially valuable error information unless `E = void`. Perhaps it would be usefull to separate this case into another class

Reply:
resolved - first and last were removed

Reply:
I've removed these functions